:recycle: Cleaning up the code and migrating to `python-jose`

acutaia opened the pull request:
Removed commented code and substitute the for loop with set operator issubset

acutaia commented:
I've cleaned a bit the code of your PR [#3305](https://github.com/tiangolo/fastapi/pull/3305)

k4black commented:
Thank you for your fixes!

k4black commented on fastapi/security/jwt.py:
```diff
@@ -102,22 +97,22 @@ def _generate_payload(
+        else:
+            if cookie:
+                cookie = str(cookie)
+            token = cookie
```
After this correction, the logical order of `places` prioritization is violated.  
It seems to me that it is more important when providing a Bearer token to use it, since it is specified more explicitly than cookies.  
Perhaps then the best way is the following
```
if bearer:
    return str(bearer.credentials)  # type: ignore
if cookie:
    return str(cookie)
return None
```

k4black commented:
Did you run local testing?  
The main drawback of python-jose is that there is no typing in this lib 😢 And we will need to invent a wheel to satisfy mypy.  
`error: Skipping analyzing 'jose': found module but no type hints or library stubs`

acutaia commented:
@k4black I've run everything except of mypi. Maybe there is a way to avoid mypi to check

k4black commented:
> @k4black I've run everything except of mypi. Maybe there is a way to avoid mypi to check
@acutaia, 
```
from jose import jwt  # type: ignore
```
Can be used  
  
Although this is not the best solution, but I can not yet think of how to do it better

acutaia commented:
I've added a config to mypi.ini in order to not check the types of jose and fixed all of mypy errors 
![Mypi_fixed](https://user-images.githubusercontent.com/48996402/121076308-8ac49580-c7d6-11eb-946d-28978cabfef7.png)

k4black commented:
Please, run
`sh scripts/tests-cov-html.sh`

k4black commented:
Thank you a lot for contribution! 🔥🔥🔥 

Last thing - Can you, please, edit README.md to match docs page? By adding 
```
* <a href="https://github.com/mpdavis/python-jose" target="_blank"><code>python-jose[cryptography]</code></a> - Required if you want to use `JwtAuth` or `JwtAuthRefresh`.
```

k4black reviewed (changes requested):
Can you, please, edit README.md to match docs page? By adding 
```
* <a href="https://github.com/mpdavis/python-jose" target="_blank"><code>python-jose[cryptography]</code></a> - Required if you want to use `JwtAuth` or `JwtAuthRefresh`.
```

acutaia commented:
Thank you for your patience 🙇🏽
